src/datastructures/mod.rs

```diff
@@ -70,3 +73,8 @@ pub use crate::datastructures::vecmatrix::VecMatrix;
 
 /// nalgebra DMatrix based implementation
 pub use crate::datastructures::nalgebra::NAMatrix;
+
+/// helper for MPI calls to send both some accumulated cost and the rank who calculated it
+#[derive(Default, Clone, Copy, Equivalence)]
+#[cfg(feature = "mpi")]
+pub struct MPICostRank(pub f64, pub i32);
```

Comment:
I suggest using the mpi `Rank` type, to stay as platform-independent as possible:
```suggestion
pub struct MPICostRank(pub f64, pub mpi::topology::Rank);
```